extism.go

```diff
@@ -217,6 +228,52 @@ type Manifest struct {
 	Timeout      uint64            `json:"timeout_ms,omitempty"`
 }
 
+type concreteManifest struct {
+	Wasm   []concreteWasm `json:"wasm"`
+	Memory struct {
+		MaxPages uint32 `json:"max_pages,omitempty"`
+	} `json:"memory,omitempty"`
+	Config       map[string]string `json:"config,omitempty"`
+	AllowedHosts []string          `json:"allowed_hosts,omitempty"`
+	AllowedPaths map[string]string `json:"allowed_paths,omitempty"`
+	Timeout      uint64            `json:"timeout_ms,omitempty"`
+}
+
+func (m *Manifest) UnmarshalJSON(data []byte) error {
+	tmp := concreteManifest{}
```

Comment:
oh I think this is where the stack overflow happens since UnmarshalJSON gets called recursively when it encounters the embedded type

Reply:
(but I don't think there's a way around it?)

Reply:
Yeah, other than splitting all the `Manifest` fields out into another struct (which ruins the usability from Go) there doesn't seem to be a good way to re-use them. I will sleep on it though.